streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -58,10 +60,18 @@ public KGroupedStreamImpl(final KStreamBuilder topology,
                                final String storeName) {
         Objects.requireNonNull(reducer, "reducer can't be null");
         Objects.requireNonNull(storeName, "storeName can't be null");
+        return reduce(reducer, keyValueStore(valSerde, storeName));
```

Comment:
Also need to change `keyValueStore` method to return `StateStoreSupplier<KeyValueStore>`